src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexPrefixAnalyzer.cs

```diff
@@ -11,6 +11,316 @@ namespace System.Text.RegularExpressions
     /// <summary>Detects various forms of prefixes in the regular expression that can help FindFirstChars optimize its search.</summary>
     internal static class RegexPrefixAnalyzer
     {
+        /// <summary>Cache of ToString() strings for the ASCII chars.</summary>
+        /// <remarks>The strings are lazily created on first use.</remarks>
+        private static string[]? s_asciiCharStrings;
+
+        /// <summary>Gets the ToString() string for the specified char.</summary>
+        private static string GetCharString(char ch)
```

Comment:
Should we consider moving this into `char` itself like we do for small `int`s?

Also, the helper appears to be unused.

Reply:
> Also, the helper appears to be unused.

Oops, yes, this can be deleted now. I previously had the analysis set up differently, where every node processed returned a string[], and for single characters I then needed a string for each. I revamped that to write into StringBuilders, at which point this became superfluous.